### ISSUE TITLE
Allow universal search "OR" based filters defined on model based questions to search on many columns

### ISSUE THREAD

**Issue by brendanheywood** [metabase/metabase#34108]
**Is your feature request related to a problem? Please describe.**
I have created a standard pattern for most of our dashboards where there might be say 5 questions on the dash and there might be 4 or 5 filters on different things, but to make it dirt easy to use I always try to add a universal 'Search' filter which is always the first filter.

![image](https://github.com/metabase/metabase/assets/187449/6dbfc3c4-aec3-41d8-8d35-04215ca9f8eb)

Lets say we have filters: Search, Name, Occupation, Favorite color.

For all of my SQL questions this mean that if you search for 'red' then it will find all records who have a name which contains 'red' OR occupation which contains red, or their fav color is red.

Another example is almost every column in every question I have created can conditionally expose errors. Eg if they are missing a Name then instead of a blank value it might contain the string 'ERROR: Missing name' highlighted in red. Now I can search for 'Error' in the universal search box and any records which have ERROR in any column are shown and I have a very simple actionable dashboard full of red things to go and fix.

![image](https://github.com/metabase/metabase/assets/187449/c5f19be9-b7f3-48fc-8bb9-01289cc02252)

**Describe the solution you'd like**
This has been working great for ages with SQL questions, however I'm falling in love with Models and trying to convert many of my dashboards over to use them to make them easier to keep consistent (instead of duplicating lots of sql into many SQL questions) and I don't think it is currently viable to have a search filter with OR logic applied to many filters.

I can see two solutions which could build on each other:
1) metabase introduces a boolean logic builder when setting up a filter. ie when you define a new filter and link it to the filters on each question you can select more than one 'Column to filter on'. By default these become UNION'ed filters.

2) metabase introduces a super simple 'Search' filter type. This doesn't need any inputs and automatically searches across any field which would be searchable via a text filter and wires them all up with a unioned OR based logic:

![image](https://github.com/metabase/metabase/assets/187449/44dc46f7-59d5-40c9-b357-c905e0b3096c)

There might be certain edge cases where you do not want to search on certain fields in which case you could click down into that question and add or remove columns as needed.

**Describe alternatives you've considered**
Doing all this manually using SQL questions

**How important is this feature to you?**
I think this approach to having a universal search filter dramatically makes a large set of dashboards approachable and means you can always zero in on data without much contextual knowledge of the data you are drilling into

**Additional context**

**Comment by brendanheywood** [comment]
A workaround I've had some success with is to make a new column which is an aggregate of all the other columns I want to search on. It ends up with a huge amount of text in it and then you can search on just that one column. It's functional but pretty ugly to setup.

**Comment by paoliniluis** [comment]
@brendanheywood is this the same as https://github.com/metabase/metabase/issues/13735?

**Comment by brendanheywood** [comment]
> @brendanheywood is this the same as https://github.com/metabase/metabase/issues/13735?

No, but they are somewhat related. #13735 would enable you to make a dashboard, and add manually add a filter to it which could be configured to search a bunch of fields and have any of them match. To make that work across any dashboards would be a lot of onerous and repetitive configuration and be very error prone and brittle to future changes. What I am proposing here should be as simple as a single checkbox per dashboard / question which is 'Enable search for this dashboard' and automatically wires everything up without the dashboard author knowing or caring what it was filtering on under the hood. And if the various questions changed, adding new columns etc then the high level search should automatically search across across all the data as they change with no additional changes to the dashboard configuration.
